DedicatedHostsManager/DedicatedHostEngine/DedicatedHostEngine.cs

```diff
@@ -826,5 +969,92 @@ await Policy
                 azureEnvironment);
             return (await computeManagementClient.DedicatedHostGroups.GetAsync(resourceGroupName, hostGroupName)).Id;
         }
+
+        private IList<(int fd, int numberOfHosts)> CalculatePlatformFaultDomainToHost(
+            DedicatedHostGroup dhGroup,
+            IList<DedicatedHost> existingHostsOnDHGroup,
+            string vmSku,
+            int vmInstancesRequested,
+            int vmCapacityPerHost,
+            int? platformFaultDomain)
+        {
+            var dedicatedHosts = new List<(int fd, int numberOfHosts)>();
+            var platformFaultDomains = DetermineFaultDomainsForPlacement(dhGroup.PlatformFaultDomainCount, platformFaultDomain);
+            var vmsRequiredPerFaultDomain = (int)Math.Round((decimal)vmInstancesRequested / platformFaultDomains.Length, MidpointRounding.ToPositiveInfinity);
+
+            foreach (var fd in platformFaultDomains)
+            {
+                var dhInFaultDomain = existingHostsOnDHGroup.Where(c => c.PlatformFaultDomain == fd);
+                var availableVMCapacityInFaultDomain = 0;
+                foreach (var host in dhInFaultDomain)
+                {
+                    // Existing hosts can be different sku then DH Sku determined for VM size.
+                    availableVMCapacityInFaultDomain += (int)(host.InstanceView?.AvailableCapacity?.AllocatableVMs?
+                        .FirstOrDefault(v => v.VmSize.Equals(vmSku, StringComparison.InvariantCultureIgnoreCase))?.Count ?? 0);
+                }
+
+                if (vmsRequiredPerFaultDomain > availableVMCapacityInFaultDomain)
+                {
+                    dedicatedHosts.Add((fd, (int)(Math.Round(((decimal)vmsRequiredPerFaultDomain - availableVMCapacityInFaultDomain) / vmCapacityPerHost, MidpointRounding.ToPositiveInfinity))));
```

Comment:
Maybe add an intrumentation/logging message that new Hosts were allocated?

Reply:
Information logged as suggested. Both at time of determining # of hosts required and also on successful creation of hosts.